daprdocs/content/en/operations/components/setup-bindings/supported-bindings/apns.md

```diff
@@ -28,13 +30,69 @@ spec:
         name: <SECRET>
         key: <SECRET-KEY-NAME>
 ```
+## Spec metadata fields
+
+| Field              | Required | Binding support | Details | Example |
+|--------------------|:--------:| ----------------|---------|---------|
+| development | Y | Output | Tells the binding which APNs service to use. Set to `"true"` to use the development service or `"false"` to use the production service. Default: `"true"` | `"true"` |
+| key-id | Y | Output | The identifier for the private key from the Apple Developer Portal | `"private-key-id`" |
+| team-id | Y | Output | The identifier for the organization or author from the Apple Developer Portal | `"team-id"` |
+| private-key | Y | Output| Is a PKCS #8-formatted private key. It is intended that the private key is stored in the secret store and not exposed directly in the configuration. See [here](#private-key) for more details | `"pem file"` |
+
+### Private Key
+The APNS binding will need a cryptographic private key in order to generate authentication tokens for the APNS service. 
+The private key can be generated from the Apple Developer Portal and is provided as a PKCS #8 file with the private key stored in PEM format. 
+The private key should be stored in the Dapr secret store and not stored directly in the binding's configuration file.
+
+A sample configuration file for the APNS binding is shown below:
+```yaml
+apiVersion: dapr.io/v1alpha1
+kind: Component
+metadata:
+  name: apns
+  namespace: default
+spec:
+  type: bindings.apns
+  metadata:
+  - name: development
+    value: false
+  - name: key-id
+    value: PUT-KEY-ID-HERE
+  - name: team-id
+    value: PUT-APPLE-TEAM-ID-HERE
+  - name: private-key
+    secretKeyRef:
+      name: apns-secrets
+      key: private-key
+```
+If using Kubernetes, a sample secret configuration may look like this:
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+    name: apns-secrets
+    namespace: default
+stringData:
+    private-key: |
+        -----BEGIN PRIVATE KEY-----
+        KEY-DATA-GOES-HERE
+        -----END PRIVATE KEY-----
+```
+
```

Comment:
We should add the alert shortcode here about using secrets instead of pasting secrets directly

Reply:
Hi @AaronCrawfis This is actually the yaml for creating a kubernetes secret. The private key will be there in this yaml which is then applied to k8s to create the secret.

Reply:
all yaml examples of Dapr component yaml contain secret key references.